CenterLineMetrics/CenterLineMetrics.py

```diff
@@ -0,0 +1,285 @@
+import os
+import unittest
+import logging
+import vtk, qt, ctk, slicer
+import numpy
+from slicer.ScriptedLoadableModule import *
+from slicer.util import VTKObservationMixin
+
+"""
+  CenterLineMetrics
+  This file is almost totally derived from LineProfile.py.
+  The core diameter calculation code is poked from VMTK's
+  README.md file.
+"""
+
+class CenterLineMetrics(ScriptedLoadableModule):
+  """Uses ScriptedLoadableModule base class, available at:
+  https://github.com/Slicer/Slicer/blob/master/Base/Python/slicer/ScriptedLoadableModule.py
+  """
+
+  def __init__(self, parent):
+    ScriptedLoadableModule.__init__(self, parent)
+    self.parent.title = "Centerline metrics"  # TODO: make this more human readable by adding spaces
+    self.parent.categories = ["Vascular Modeling Toolkit"]  # TODO: set categories (folders where the module shows up in the module selector)
+    self.parent.dependencies = []  # TODO: add here list of module names that this module requires
+    self.parent.contributors = ["SET (Hobbyist)"]  # TODO: replace with "Firstname Lastname (Organization)"
+    self.parent.helpText = """
+This module plots average diameters around a VMTK centerline model. It is intended for non-bifurcated centerlines.
+"""  # TODO: update with short description of the module
+    self.parent.helpText += self.getDefaultModuleDocumentationLink()  # TODO: verify that the default URL is correct or change it to the actual documentation
```

Comment:
Replace this with actual link to the module documentation

Reply:
Will search how to do it all.

Reply:
Done.